app/src/main/java/org/p2p/wallet/striga/onboarding/StrigaOnboardingFragment.kt

```diff
@@ -0,0 +1,187 @@
+package org.p2p.wallet.striga.onboarding
+
+import androidx.activity.addCallback
+import androidx.core.net.toUri
+import androidx.core.view.isVisible
+import android.content.Intent
+import android.os.Bundle
+import android.text.method.LinkMovementMethod
+import android.view.View
+import org.koin.android.ext.android.inject
+import org.p2p.core.common.DrawableContainer
+import org.p2p.core.common.TextContainer
+import org.p2p.uikit.components.finance_block.FinanceBlockCellModel
+import org.p2p.uikit.components.icon_wrapper.IconWrapperCellModel
+import org.p2p.uikit.components.left_side.LeftSideCellModel
+import org.p2p.uikit.components.right_side.RightSideCellModel
+import org.p2p.uikit.utils.SpanUtils
+import org.p2p.uikit.utils.image.ImageViewCellModel
+import org.p2p.uikit.utils.image.bind
+import org.p2p.uikit.utils.text.TextViewCellModel
+import org.p2p.uikit.utils.text.bind
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.repository.Country
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentStrigaOnboardingBinding
+import org.p2p.wallet.intercom.IntercomService
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.viewbinding.getColor
+import org.p2p.wallet.utils.viewbinding.getDrawable
+import org.p2p.wallet.utils.viewbinding.getString
+import org.p2p.wallet.utils.viewbinding.viewBinding
+
+class StrigaOnboardingFragment :
+    BaseMvpFragment<StrigaOnboardingContract.View, StrigaOnboardingContract.Presenter>(
+        R.layout.fragment_striga_onboarding
+    ),
+    StrigaOnboardingContract.View {
+
+    companion object {
+        fun create(): StrigaOnboardingFragment = StrigaOnboardingFragment()
+    }
+
+    override val presenter: StrigaOnboardingContract.Presenter by inject()
+
+    private val binding: FragmentStrigaOnboardingBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        handleBackPress()
+        bindHelpText()
+
+        binding.buttonContinue.setOnClickListener {
+            presenter.onClickChangeCountry()
+        }
+        binding.textViewPoweredBy.setOnClickListener {
+            Intent(Intent.ACTION_VIEW, getString(R.string.striga_powered_by_url).toUri())
+                .also { startActivity(it) }
+        }
+    }
+
+    override fun navigateNext() {
+        // TODO: navigate to next screen
+    }
+
+    override fun setCurrentCountry(country: Country) {
+        mapCountryView(country.name, country.flagEmoji)
+    }
+
+    override fun setAvailabilityState(state: StrigaOnboardingContract.View.AvailabilityState) {
+        when (state) {
+            StrigaOnboardingContract.View.AvailabilityState.Available -> handleAvailableState(state)
+            StrigaOnboardingContract.View.AvailabilityState.Unavailable -> handleUnavailableState(state)
+        }
+
+        binding.image.animate()
```

Comment:
wow, whats that?
also
image -> imageViewImage

Reply:
since image sets dynamically, at first launch it appears a bit jerky, so I decided to show it with alpha